### PR TITLE
Teach PathValidator a global condition on edges

### PR DESCRIPTION
**Pull request by markuspf**
### Scope & Purpose

The PathValidator class can already enforce a global condition on all vertices; this patch also allows it to filter all edges on a path.

### CODE REVIEW

**Comment by markuspf** [comment]
@mchacki suggested a better change that enables use of indices right away; I will open another PR with this alternative code. In case the alternative code does not work out (yet), this PR can still be used.

**Comment by markuspf** [comment]
Unfortunately @mchacki 's suggestion does not work, as the code makes assumptions that cannot be fulfilled. 

For the time being this code here will need to be used.

File: arangod/Graph/PathManagement/PathValidator.cpp [hunk 1]
```diff
@@ -302,6 +302,22 @@ auto PathValidator<ProviderType, PathStore, vertexUniqueness, edgeUniqueness>::
...
+
+      _provider.addEdgeToBuilder(step.getEdge(), edgeBuilder);
+      bool satisfiesCondition =
+          evaluateExpression(edgeExpr, edgeBuilder.slice());
+      if (!satisfiesCondition) {
+        return ValidationResult{ValidationResult::Type::FILTER_AND_PRUNE};
+      }
+    } else {
+      // TODO: at the moment we smile and wave...
+    }
```
Comment (hkernbach) [inline arangod/Graph/PathManagement/PathValidator.cpp:318]: Would just remove that block - as it does nothing right now.

Comment (markuspf) [reply arangod/Graph/PathManagement/PathValidator.cpp:318]: I put that comment in there after debugging tests to make it clear that invalid edges just pass this check.

Tbh I don't entirely know what an "invalid" edge is and under which circumstances they would end up here. If I ADB_PROD_ASSERT() here, I think everything explodes

**Pull request merged**